alignedbuff/alignedbuff.go

```diff
@@ -118,6 +129,40 @@ func (a *AlignedBuff) Uint64() (uint64, error) {
 	return v, nil
 }
 
+// Int32 unmarshals an int32 in native endianess and alignment. It returns
+// ErrEOF when trying to read beyond the payload.
+func (a *AlignedBuff) Int32() (int32, error) {
+	if err := a.alignCheckedRead(int32AlignMask); err != nil {
+		return 0, err
+	}
+	v := binaryutil.Int32(a.data[a.pos : a.pos+4])
+	a.pos += 4
+	return v, nil
+}
+
+// String unmarshals a null terminated string
+func (a *AlignedBuff) String() (string, error) {
+	len := 0
+	for {
+		if a.data[a.pos+len] == 0x00 {
+			break
+		}
+		len++
+	}
+
+	v := binaryutil.String(a.data[a.pos : a.pos+len])
```

Comment:
presumably this ignores the '\0' at the end ?

Reply:
Dunno, these are upstream changes